ionic2/src/components/poi-card/poi-card.component.ts

```diff
@@ -53,13 +54,20 @@ export class POICardComponent implements OnInit {
 
     // Load Pokemon for given pokemonId
     if (poi instanceof Sighting) {
-      const sighting = <Sighting>poi;
+      const sighting = <Sighting> poi;
       this.loadPokemon = this.apiService
                              .getPokemonById(sighting.pokemonId)
                              .subscribe(pokemon => {
                                 this.slideState = 'visible';
                                 this.pokemon = pokemon;
-                                this.changeDetectorRef.detectChanges();
+                              });
+    } else if (poi instanceof Prediction) {
+      const prediction = <Prediction> poi;
+            this.loadPokemon = this.apiService
+                             .getPokemonById(prediction.pokemonId)
+                             .subscribe(pokemon => {
+                                this.slideState = 'visible';
+                                this.pokemon = pokemon;
```

Comment:
I was pretty convinced it was over-engineering, but things are pretty now: https://github.com/PokemonGoers/Catch-em-all/pull/148/commits/2b6b5ccbe97e0e821df3364b8ffb91bcf4773cf3